File: elixir/wordy/wordy.exs
defmodule Wordy do

  @doc """
  Calculate the math problem in the sentence.
  """
  @spec answer(String.t) :: integer
  def answer(question) do
    question
    |> String.replace(~r/\?/, "")
    |> String.split(" ")
    |> do_answer(0)
  end

  defp do_answer([], result), do: result

  defp do_answer(["What", "is" | tail], result) do
    do_answer(tail, result)
  end

  defp do_answer(["plus", num | tail], result) do
    do_answer(tail, result + String.to_integer(num))
  end

  defp do_answer(["minus", num | tail], result) do
    do_answer(tail, result - String.to_integer(num))
  end

  defp do_answer(["multiplied", "by", num | tail], result) do
    do_answer(tail, result * String.to_integer(num))
  end

  defp do_answer(["divided", "by", num | tail], result) do
    do_answer(tail, div(result, String.to_integer(num)))
  end

  defp do_answer(["raised", "to", "the", num, "power" | tail], result) do
    {pow, _} = Integer.parse(num)

    do_answer(tail, :math.pow(result, pow))
  end

  defp do_answer([head | tail], result) do
    case Integer.parse(head) do
      :error -> raise ArgumentError
      {num, _} -> do_answer(tail, num)
    end
  end

end

File: go/wordy/wordy.go
package wordy

import (
	"strconv"
	"strings"
)

func Answer(question string) (int, bool) {
	q := strings.TrimPrefix(strings.TrimSuffix(question, "?"), "What is ")

	plus := func(a, b int) int { return a + b }
	minus := func(a, b int) int { return a - b }
	times := func(a, b int) int { return a * b }
	divide := func(a, b int) int { return a / b }

	tokens := strings.Split(q, " ")
	sum := 0

	val, err := strconv.Atoi(tokens[0])
	if err != nil {
		return -1, false
	}

	sum = val

	for i := 1; i < len(tokens); {
		fn := plus
		switch tokens[i] {
		case "plus":
			fn = plus
			i++
		case "minus":
			fn = minus
			i++
		case "multiplied":
			fn = times
			i += 2
		case "divided":
			fn = divide
			i += 2
		}

		val, err := strconv.Atoi(tokens[i])
		if err != nil {
			return -1, false
		}

		sum = fn(sum, val)
		i++
	}

	return sum, true
}
